src/Frame.cc

```diff
@@ -27,12 +29,56 @@ Frame::Frame(int arg_size, const BroFunc* func, const val_list* fn_args)
 	Clear();
 	}
 
+Frame::Frame(const Frame* other)
+	{
+	this->size = other->size;
+	this->frame = other->frame;
+	this->function = other->function;
+	this->func_args = other->func_args;
+
+	this->next_stmt = 0;
+	this->break_before_next_stmt = false;
+	this->break_on_return = false;
+	this->delayed = false;
+
+	if ( other->trigger )
+		Ref(other->trigger);
+
+	for (int i = 0; i < size; i++)
+	  if (frame[i])
+	    Ref(frame[i]);
```

Comment:
@jsiwek I could use a sanity check on reference counting here if you get a chance. 

This constructor is intended to create a "view" of the input frame. ClosureFrame uses it so that it can route most methods to the underlying frame without having to add a decent amount of extra code to do that. I'm Refing everything I'm viewing because when ClosureFrame is taken apart it Unrefs it all. I wasn't getting any trouble without the Ref calls on all the elements though from my tests so I wonder if it is actually needed.